### PR TITLE
fix(types): update Hit type

### PR DESCRIPTION
**Pull request by tkrugg**
**Summary**

This type spec is copy pasted from the one we currently have in InstantSearch.

https://github.com/algolia/instantsearch.js/blob/65911504b9ae0e2596c75f0ae74133985dc79850/src/types/instantsearch.ts#L20-L71

Generally copy pasting is not a good idea but [this comment](https://github.com/algolia/angular-instantsearch/pull/553/files#r284135258) makes the point that since it's mostely Search API spec, it may not change much.

**Results**
N/A

### CODE REVIEW

File: src/instantsearch/instantsearch.ts [hunk 1]
```diff
@@ -135,8 +135,57 @@ export type SearchResponse = {
...
+    userScore: number;
+    matchedGeoLocation?: {
+      lat: number;
+      lng: number;
+      distance: number;
+    };
+  };
+  _distinctSeqID?: number;
+  __position: number;
+  __queryID?: string;
```
Comment (Haroenv) [inline src/instantsearch/instantsearch.ts]: are these last two also used like this in Angular InstantSearch?

Comment (tkrugg) [reply src/instantsearch/instantsearch.ts]: They're not but an Angular InstantSearch user might need them, typically in a template when trying to pass it in a URL.

```html
<ais-hits>
  <ng-template let-hits="hits">
    <a 
       *ngFor="let hit of hits"
       href="product.html?queryID={{hit.__queryID}}"
     >
        ...
    </a>
  </ng-template>
</ais-hits>
```

Other than that, I don't want to start diverging for the types we have in InstantSearch. Does that make sense?

Comment (Haroenv) [reply src/instantsearch/instantsearch.ts]: makes sense, my question was more _if_  this property is available, which seems to be true

Comment (tkrugg) [reply src/instantsearch/instantsearch.ts]: ah yes it is, it's exposed by connectHits and connectInfiniteHits

**Pull request merged**